client/components/prompt-input/PromptInputForm.tsx

```diff
@@ -84,18 +106,8 @@ const PromptInputForm = () => {
     },
   });
 
-  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    setInputValue(event.target.value);
-  };
-
-  const handleSubmit = () => {
-    newUserInputMutation.mutate({ input: inputValue });
-  };
-
-  const handleCheckboxClick = (event: React.ChangeEvent<HTMLInputElement>) => {
-    const newCheckedState = event.target.checked;
-    setIsChecked(newCheckedState);
-    localStorage.setItem('isChecked', newCheckedState.toString());
+  const onSubmit = (userInput: UserInputPayload) => {
```

Comment:
I think we might need a `reset()` int he onSuccess handler to clear the form state after submssion.

Reply:
Good catch. I also added an isSubmitted value to the GenerateButton's disabled prop so inputs can't be submitted after the first submission. Makes for good UX at least until I finish the Loading state UI.